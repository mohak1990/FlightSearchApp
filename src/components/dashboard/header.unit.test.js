import React from 'react';
import Header from './header';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';


test('Snapshot testing for header component', () => {
  const component = renderer.create(
    <Header flights = {[]} date = {new Date("05/24/2016")} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('render header correctly', () => {
    const HeaderComponent = mount(<Header flights = {[]} />).render();
    expect(HeaderComponent.hasClass('dashboard--header')).toBeTruthy();
});

it('render header correctly for return flights', () => {
    const HeaderComponent = mount(<Header flights = {[]} isReturnFlight = {true} />).render();
    expect(['dashboard--header', 'dashboard--returnFlight'].every(c => HeaderComponent.hasClass(c))).toBeTruthy();
});
