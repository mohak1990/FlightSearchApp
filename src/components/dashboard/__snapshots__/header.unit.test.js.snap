// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing for header component 1`] = `
<div
  className="dashboard--header"
>
   to 
  <div
    className="dashboard--header--info"
  >
    0
     flights found    
    05/24/2016
  </div>
</div>
`;
